proto/buf/alpha/registry/v1alpha1/recommendation.proto

```diff
@@ -40,6 +70,20 @@ service RecommendationService {
   rpc RecommendedRepositories(RecommendedRepositoriesRequest) returns (RecommendedRepositoriesResponse);
   // RecommendedTemplates returns a list of recommended templates.
   rpc RecommendedTemplates(RecommendedTemplatesRequest) returns (RecommendedTemplatesResponse);
+  // ListRecommendedRepositories returns a list of recommended repositories that user have access to.
+  rpc ListRecommendedRepositories(ListRecommendedRepositoriesRequest) returns (ListRecommendedRepositoriesResponse);
+  // ListRecommendedTemplates returns a list of recommended templates that user have access to.
+  rpc ListRecommendedTemplates(ListRecommendedTemplatesRequest) returns (ListRecommendedTemplatesResponse);
+  // ListRecommendations returns a list of recommendations of a resource type.
+  rpc ListRecommendations(ListRecommendationsRequest) returns (ListRecommendationsResponse);
+  // AddRecommendation add a recommendation to the server.
+  rpc AddRecommendation(AddRecommendationRequest) returns (AddRecommendationResponse);
+  // UpdateRecommendation update a recommendation in the server.
+  rpc UpdateRecommendation(UpdateRecommendationRequest) returns (UpdateRecommendationResponse);
+  // UpdateRecommendationsRanks update the ranking of the recommendations of a resource type in the server.
+  rpc UpdateRecommendationsRanks(UpdateRecommendationsRanksRequest) returns (UpdateRecommendationsRanksResponse);
+  // DeleteRecommendation delete a recommendation in the server.
+  rpc DeleteRecommendation(DeleteRecommendationRequest) returns (DeleteRecommendationResponse);
```

Comment:
If we treated recommendations as immutable, we could reduce this to just two endpoints; `AddRecommendations` and `DeleteRecommendations`. If we make it `AddRecommendedRepositories` and `DeleteRecommendedRepositories` and corresponding methods for Templates, we can remove the `Resource` type and enum. I appreciate the value of reducing the number of RPCs, but I really kinda hate the enum-to-determine-type :P. I'm obviously open to change my mind, I think we discussed the explosion of RPCs before.

Reply:
I guess we still want to keep the update rank endpoint, otherwise user will need to delete all recommendations and add them again if they want to add some recommendation in the middle of the list? This also means two/three extra endpoints needed per new recommendation type in the future, though it might be too early to think about it.

Reply:
Yes, but the user doesn't have to be exposed to the method calls that perform the deletion/addition. We could actually use a `SetRecommendedRepositories` instead that could just overwrite the contents every time, no need to delete. Then the frontend could have a nice drag and drop table with a `Save` button or something to update them all at once. Would be nice to get the input from @timostamm who will most likely implement it!

Reply:
the drag and drop table is actually what I was thinking so that I come up with `UpdateRecommendationsRanks`, but put the add and delete as separated actions. Do you mean no `add`, no `delete`, just `set` for each type of resources?

Reply:
Perhaps, I thought we might still have a `Delete` but maybe you can just `Set` it to an empty list!

Reply:
sounds good for me, @timostamm what do you think?

Reply:
> Yes, but the user doesn't have to be exposed to the method calls that perform the deletion/addition. We could actually use a `SetRecommendedRepositories` instead that could just overwrite the contents every time, no need to delete. 

I agree that this should be sufficient. Handling in the frontend would be very straight-forward. I expect payload sizes to be so small that it would not make a noticeable difference even with many dozen entries.

Reply:
thats cool, I guess we don't need any `enum`/`oneof` at all too in this case

Reply:
also @timostamm if payload sizes is not a concern for `set`, do we still need to do paging for `ListRecommendedRepositories` and `ListRecommendedTemplates`? my only concern is I don't know how many recommendation items user will set.

Reply:
I don't think it makes sense to recommend more than 10 repositories to the user (without some kind of categorization), and even less for templates (per registry). Where we set the limit is something we should discuss, but from my point of view, pagination is not necessary.

Reply:
cool, then I will also remove the pagination, but will keep the new `ListRecommendedRepositories` and `ListRecommendedTemplates` endpoints so that we can gracefully migrate to the new configurable recommendation.